routes/upload.py

```diff
@@ -41,12 +43,20 @@ def handle(self, app, item):
             # if user does not select file, browser also
             # submit an empty part without filename
             if file.filename == '' or not file.filename.lower().endswith('.zip'):
-                logging.error('No filename or invalid filename')
+                logging.error('No driver filename or invalid filename')
+                self.setAbort(405)
+                return
+        elif item == 'config':
+            # if user does not select file, browser also
+            # submit an empty part without filename
+            if file.filename == '' or not file.filename.lower().endswith('.tar.gz'):
```

Comment:
Probably should use magic module here if possible since file extensions can be ... unreliable ;-) and since this is sent on to possibly `tar` with minimal checks, we could create a bigger issue. Should also ideally validate that the .tar.gz file contains a backup, but perhaps the restore tool does that?

Reply:
Yes, the restore tool does that.   It could also check a restored backup to see if the typical files and directories are there.  Is that what you had in mind?  I started going down that path, and then it seemed I was re-writing the whole load config section of photoframe.   I thought it would be better just to try to load the config.

Reply:
I think just checking that the bare minimal files are there including the version number (I think we spoke about this a while ago 😉 ) should be enough. Just so people don't load a non-relevant TGZ file.

Reply:
I've checked against the 2 directories and 2 top-level json files - not the services/services.json file.  I think we're protected from random TGZ files.  However, I'm wondering what a valid configuration upload is or should be?   At this point it seems that configurations are pretty mobile across all releases and devices - meaning I can see trying to move a config across all sorts of systems - especially rather than trying to re-do the auth credentials for google. ;-)    I have used the same config file to jumpstart systems on various branches and hardware.

I have added code to create a version.json file in settings.  It includes the linux and python versions, as well as git origin branch and commit.  That says exactly what saved the settings.  At some point I think it may become the case that some settings cannot be restored to some other versions, and I wonder if strict checking against the same origin/branch/commit is good.   For example, if I were concerned about trying some settings in a new system, I'd likely backup the existing setting to /boot/ and then try to upload the new ones.  If they should fail, I could restore from /boot/.   

I realize that this does not include a separate version number (i.e. "1"), it would be dead-easy to add, but I'm wondering what that would provide that the origin/branch/commit do not?   Thanks!